prolog/main.pl

```diff
@@ -53,15 +54,56 @@
 				true
 			; SelectedText = 1 ->
 				tutorial()))).
-	
 
-gameLoop(State) :-
-	get_single_char(Input),
-	true.
+:- dynamic playerPosition/2.
+:- dynamic quitGame/2.
+:- dynamic endGame/2.
 
+movement(119, -1, 0).
+movement(115, 1, 0).
+movement(100, 0, 1).
+movement(97, 0, -1).
+
+getUserAction() :- endGame(1).
+getUserAction() :-
+	endGame(0),
+	get_single_char(UserAction),
+	applyUserAction(UserAction).
+
+applyUserAction(UserAction) :-
+	playerPosition(X, Y),
+	movement(UserAction, MovX, MovY),
+	NewX is (X + MovX),
+	NewY is (Y + MovY),
+	retract(playerPosition(X, Y)),
+	asserta(playerPosition(NewX, NewY)),
+	getUserAction().
+
+applyUserAction(27) :- 
```

Comment:
Shouldn't come first then "applyUserAction(UserAction) :-" to avoid that UserAction use 27 as a valid value?

Reply:
applyUserAction(UserAction) clause will be only true if movement(UserAction, MovX, MovY) is true, note that movement(27, MovX, MovY) is not true.

Reply:
It make sense. I still prefer it comming first, but it doesn't matter :) Great work!